src/Rules/BannedNodesRule.php

```diff
@@ -73,4 +79,22 @@ public function processNode(Node $node, Scope $scope): array
 
         return [sprintf('Should not use node with type "%s", please change the code.', $type)];
     }
+
+    /**
+     * Strip leading slashes from function names.
+     *
+     * php-parser makes the same normalization.
+     *
+     * @param  array<string, mixed>  $bannedNodes
+     * @return array<string>
+     */
+    protected function normalizeFunctionNames(array $bannedNodes): array
```

Comment:
From the method name to see if it should not have a return value and there is no need to pass parameters.

Reply:
Mutating parameters is neither common nor good practice in PHP and requires passing by reference for array. Seems bad.

Reply:
Wouldn't the following be better practice?
```PHP
    public function __construct(array $bannedNodes)
    {
        $this->bannedNodes     = array_column($bannedNodes, null, 'type');
        $this->bannedFunctions = self::normalizeFunctionNames
    }
  protected function normalizeFunctionNames(): void
    {
        $this->bannedFunctions =  array_map(
            static function (string $function): string {
                return ltrim($function, '\\');
            },
            $this->bannedNodes['Expr_FuncCall']['functions'] ?? []
        );
    }
```

Reply:
I think it would be much worse, the signature `(): void` expresses no intent.